pkg/operator/controller/externaldns/deployment.go

```diff
@@ -322,12 +322,17 @@ func externalDNSAnnotationsChanged(current, expected, updated *appsv1.Deployment
 	changed := false
 	if current.Annotations == nil {
 		updated.Annotations = expected.Annotations
-		return true
+		changed = true
```

Comment:
As discussed, I don't think we need to keep the annotation in the deployment's metadata.

Reply:
Removed deployment's annotation.